src/chains/ethereum/ethereum/tests/api/eth/feeHistory.test.ts

```diff
@@ -0,0 +1,525 @@
+            const blockCount = "0x0";
+            const newestBlock = "0x0";
+            const feeHistory = await provider.send("eth_feeHistory", [
+              blockCount,
+              newestBlock,
+              []
+            ]);
+
+            assert.strictEqual(feeHistory.oldestBlock, "0x0");
+            assert.strictEqual(feeHistory.baseFeePerGas, undefined);
+            assert.strictEqual(feeHistory.gasUsedRatio, null);
+            assert.strictEqual(feeHistory.reward, undefined);
+          });
+          describe("newestBlock", () => {
+            it("throws if header is not found", async () => {
+              const blockCount = "0x1";
+              await assert.rejects(
+                provider.send("eth_feeHistory", [
+                  blockCount,
+                  headerNotFoundBlock,
+                  []
+                ]),
+                new Error(ERROR_HEADER_NOT_FOUND)
+              );
+            });
+          });
+          describe("rewardPercentile", () => {
+            it("returns undefined if no percentiles are given", async () => {
+              const blockCount = "0x1";
+              const newestBlock = "0x2";
+              const feeHistory = await provider.send("eth_feeHistory", [
+                blockCount,
+                newestBlock,
+                []
+              ]);
+
+              assert.strictEqual(feeHistory.reward, undefined);
+            });
+          });
+        });
+      });
+      describe("response", () => {
+        describe("oldestBlock", () => {
+          it("ignores blocks that do not exist (blockCount > newestBlock)", async () => {
+            const blockCount = "0xa";
+            const newestBlock = "0x2";
+            const feeHistory = await provider.send("eth_feeHistory", [
+              blockCount,
+              newestBlock,
+              []
+            ]);
+
+            assert.strictEqual(feeHistory.oldestBlock, genesisBlock);
+            assert.strictEqual(feeHistory.baseFeePerGas.length, 4); // blocks found + 1
+            assert.strictEqual(feeHistory.gasUsedRatio.length, 3);
+          });
+          it("newestBlock = oldestBlock when blockCount === 1", async () => {
+            const blockCount = "0x1";
+            const newestBlock = "0x2";
+            const feeHistory = await provider.send("eth_feeHistory", [
+              blockCount,
+              newestBlock,
+              []
+            ]);
+
+            assert.strictEqual(feeHistory.oldestBlock, newestBlock);
+          });
+          it("oldestBlock = newestBlock - blockCount +1", async () => {
+            const blockCount = "0x1";
+            const newestBlock = "0x2";
+            const feeHistory = await provider.send("eth_feeHistory", [
+              blockCount,
+              newestBlock,
+              []
+            ]);
+
+            assert.strictEqual(feeHistory.oldestBlock, newestBlock);
+          });
+        });
+      });
+      describe("gasUsedRatio", () => {
```

Comment:
Should this describe block (and the others that are named after one of the response parameters) be inside of the `"response"` block?

Reply:
:+1: